src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,6 @@
+package core.basesyntax;
+
+public interface Figure extends AreaCalculable, Drawable {
```

Comment:
As I said, Having `Figure` and `AbstractFigure` is redundant in your case. Combine them into one abstract class (`Figure`).

Reply:
I tried:
package core.basesyntax;

public abstract class Figure implements AreaCalculable, Drawable {
    private final String color;

    public Figure(String color) {
        this.color = color;
    }

    @Override
    public abstract double calculateArea();

    @Override
    public void draw() {
        System.out.printf("Drawing a %s figure with color %s\n", getClass().getSimpleName(), color);
    }

    public String getColor() {
        return color;
    }
}

but i have problem with superclass

Reply:
I answered in the main comment below, ask questions in chat next time (`pl_jv0_fx_chat`), because mentors do not see these questions until the student requests a review.